warehouse/query-core/src/main/java/datawave/query/data/parsers/EventKey.java

```diff
@@ -30,10 +31,18 @@ public class EventKey implements KeyParser {
     private String value;
     private String field;
 
+    private static final Logger log = Logger.getLogger(EventKey.class);
+
     @Override
     public void parse(Key k) {
-        clearState();
-        this.key = k;
+        if (this.key != null && this.key.equals(k)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Same key given to parse call. Skipping re-load: " + this.key.toString());
```

Comment:
If we're using this key for debugging, you might want to use `key.toStringNoTruncate()`. By default, Accumulo's Key.toString() will truncate making it hard to find the problematic key.

Reply:
alright, will update this.